controllers/spinner.py

```diff
@@ -58,53 +28,14 @@ def select_task(self):
             self.task = "position"  # TODO Implement FMS Connection
             self.required_colour = "R"
 
-        if self.task == "position":
+        elif self.task == "position":
             self.required_colour = "R"
             self.state = "position"
-            # self.next_state("position")
-        else:
-            self.state = "rotation"
-            # self.next_state("rotation")
-
-    # @state()
-    def position(self):
-        distance = self.get_wheel_dist()
-        self.spinner.spinner_motor.set(distance * self.WHEEL_SPIN_FACTOR)
+            self.spinner.go_to_colour("R")
 
-    # @state()
-    def rotation(self):
-        pass
+        elif self.task == "rotation":
+            self.state = "rotation"
+            self.spinner.do_rotation_control()
 
     def execute(self):
-        if self.state == "position":
-            self.position()
-        elif self.state == "rotation":
-            self.rotation()
-        elif self.state == "select_task":
-            self.select_task()
-
-    def get_current_colour(self):
-        sensed_colour, _ = self.spinner.read_colour()
-
-        distances = {}
-        for col_name, col_value in self.WHEEL_COLOURS.items():
-            distances[col_name] = col_value.dist(sensed_colour)
-
-        if (
-            distances[min(distances, key=lambda col: distances[col])]
-            < self.MAX_COLOUR_DIST
-        ):
-            return min(distances, key=lambda col: distances[col])
-
-    def get_wheel_dist(self):
-        current_colour = self.get_current_colour()
-        if current_colour == None:
-            return 0
-        distance = (
-            self.WHEEL_ORDER.index(self.required_colour)
-            + self.COLOUR_OFFSET
-            - self.WHEEL_ORDER.index(current_colour)
-        )
-        if distance > 2:
-            return distance - 4
-        return distance
+        pass
```

Comment:
This is not a component and probably doesn't need an execute function, especially an empty one.

Reply:
I deleted the execute for the controller